backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/command/InterimInvoiceFlatrateTermCreateCommand.java

```diff
@@ -0,0 +1,248 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contacts.invoice.interim.command;
+
+import ch.qos.logback.classic.Level;
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.service.IBPartnerDAO;
+import de.metas.common.util.CoalesceUtil;
+import de.metas.contacts.invoice.interim.InterimInvoiceFlatrateTerm;
+import de.metas.contacts.invoice.interim.InterimInvoiceSettings;
+import de.metas.contacts.invoice.interim.InterimInvoiceSettingsId;
+import de.metas.contacts.invoice.interim.service.IInterimInvoiceFlatrateTermBL;
+import de.metas.contacts.invoice.interim.service.IInterimInvoiceFlatrateTermDAO;
+import de.metas.contacts.invoice.interim.service.IInterimInvoiceFlatrateTermLineDAO;
+import de.metas.contacts.invoice.interim.service.IInterimInvoiceSettingsDAO;
+import de.metas.contracts.ConditionsId;
+import de.metas.contracts.FlatrateTermId;
+import de.metas.contracts.IFlatrateDAO;
+import de.metas.contracts.model.I_C_Flatrate_Conditions;
+import de.metas.contracts.model.I_C_Flatrate_Term;
+import de.metas.contracts.model.X_C_Flatrate_Conditions;
+import de.metas.contracts.process.FlatrateTermCreator;
+import de.metas.inout.IInOutDAO;
+import de.metas.inout.InOutAndLineId;
+import de.metas.invoicecandidate.api.IInvoiceCandidateHandlerBL;
+import de.metas.logging.LogManager;
+import de.metas.money.CurrencyId;
+import de.metas.order.IOrderDAO;
+import de.metas.order.OrderLine;
+import de.metas.order.OrderLineId;
+import de.metas.order.OrderLineRepository;
+import de.metas.product.IProductDAO;
+import de.metas.product.ProductId;
+import de.metas.uom.IUOMConversionBL;
+import de.metas.uom.UomId;
+import de.metas.util.Check;
+import de.metas.util.Loggables;
+import de.metas.util.Services;
+import lombok.Builder;
+import lombok.NonNull;
+import org.adempiere.ad.trx.api.ITrxManager;
+import org.adempiere.exceptions.AdempiereException;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_C_BPartner;
+import org.compiere.model.I_C_Order;
+import org.compiere.model.I_M_Product;
+import org.slf4j.Logger;
+
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Properties;
+
+public class InterimInvoiceFlatrateTermCreateCommand
+{
+	private static final Logger logger = LogManager.getLogger(InterimInvoiceFlatrateTermCreateCommand.class);
+
+	@NonNull
+	private final Properties ctx;
+	@NonNull
+	private final BPartnerId bpartnerId;
+
+	@NonNull
+	private final ProductId productId;
+	@NonNull
+	private final Timestamp dateFrom;
+	@NonNull
+	private final Timestamp dateTo;
```

Comment:
as a general rule:
* consider using Instant instead of Timestamp
* Quantity instead of BigDecimal qty
* Money/Amount instead of BigDecimal price/amt/cost
* Duration instead of int duration
* etc

Reply:
done.